File: 0050-powx-n/0050-powx-n.cpp
class Solution {
public:
    // 2^10 4^5 4*4^4 16^2 256^1*2 1024

    double myPow(double x, int n) {
        long long nn = n;
        if (nn < 0)
            nn = -nn;
        double ans = 1.0;
        while (nn > 0) {
            if (nn & 1) {
                ans = ans * x;
                cout << ans << endl;
            }

            x *= x;
            nn >>= 1;
        }
        return n < 0 ? 1 / ans : ans;
    }
};

File: 0050-powx-n/0050-powx-n.js
/**
 * @param {number} x
 * @param {number} n
 * @return {number}
 */
var myPow = function (x, n) {
    let result = 1
    let power = Math.abs(n)
    while (power) {
        if (power % 2 === 1) {
            result *= x
        }
        x *= x
        power = Math.floor(power / 2)
    }
    return n > 0 ? result : 1 / result
};